test/arithmetic/fixed_point/fixed_point_value_test.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// 2024 September 24
+// Authors:
+//  Max Korbel <max.korbel@intel.com>
+//  Desmond A Kirkpatrick <desmond.a.kirkpatrick@intel.com
+//  Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'dart:math';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('FXP: constructor', () {
+    const m = 4;
+    const n = 3;
+
+    final corners = [
+      // sign, integer, fraction
+      ['00000000', 0, 0, 0],
+      ['01111111', 0, pow(2, m)-1, pow(2, n)-1],
+      ['10000001', 1, pow(2, m)-1, pow(2, n)-1],
+      ['11101100', 1, 2, 4],  // -2.5
+    ];
+    for (var c = 0; c < corners.length; c++) {
+      final val = LogicValue.ofString(corners[c][0] as String);
+      final fxp = FixedPointValue(
+        sign: LogicValue.ofInt(corners[c][1] as int, 1),
```

Comment:
I don't understand the use of 'as int':  I would exect:
integer: LogicValue.ofInt(corners[c][2], width: m),
fraction: LogicValue.ofInt(corners[c][3], width:n),

Reply:
ofInt requires integer yet the list elements are objects by default and object cannot be assigned to an int. I can work around this by defining a list of type dynamic `final corners = <dynamic>[...` then Dart analysis still reports this as a problem `avoid_dynamic_calls`. Is there a cleaner way?

Reply:
You can avoid dynamic types by using a List of tuples as follows

  ```dart
  final corners = [
      // result, sign, integer, fraction
      ('00000000', 0, 0, 0),
      ('01111111', 0, pow(2, m).toInt() - 1, pow(2, n) - 1),
      ('10000001', 1, pow(2, m).toInt() - 1, pow(2, n) - 1),
      ('11101100', 1, 2, 4), // -2.5
    ];
    for (var c = 0; c < corners.length; c++) {
      final val = LogicValue.ofString(corners[c].$1);
      final fxp = FixedPointValue(
          sign: LogicValue.ofInt(corners[c].$2, 1),
          integer: LogicValue.ofInt(corners[c].$3, m),
          fraction: LogicValue.ofInt(corners[c].$4 as int, n),
      );
      expect(val, fxp.value);
    }
```